R/plotAbundanceDensity.R

```diff
@@ -0,0 +1,163 @@
+#'
+#' @name plotAbundanceDensity
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' tse <- microbiomeDataSets::atlas1006()
+#' # Plots the abundances of 100 most abundant taxa
+#' plotAbundanceDensity(tse, abund_values = "counts", n = 100)
+#' # Counts relative abundances
+#' tse <- transformSamples(tse, method = "relabundance")
+#' # Plots the relative abundance of 50 msot abundant taxa. 
+#' # "nationality" information is used to color the points. 
+#' plotAbundanceDensity(tse, abund_values = "relabundance", colour_by = "nationality")
+#' 
+NULL
+
+#' @rdname plotAbundanceDensity
+#' @export
+setGeneric("plotAbundanceDensity", signature = c("object"),
+           function(object,
+                    abund_values = "counts",
+                    n = 50,
+                    colour_by = NULL, 
+                    ...)
+             standardGeneric("plotAbundanceDensity"))
+
+#' @rdname plotAbundanceDensity
+#' @export
+setMethod("plotAbundanceDensity", signature = c(object = "SummarizedExperiment"),
+    function(object,
+             abund_values = "counts",
+             n = 50, 
+             colour_by = NULL, 
+             ...){
+        ############################# Input Check ##############################
+        # Checks abund_values
+        .check_assay_present(abund_values, object)
+        # Checks n
+        if( !(n%%1==0 && n>0) ){
+            stop("'n' must be a positive integer.", call. = FALSE)
+        }
+        # Checks colour_by
+        if( !is.null(colour_by) &&
+            (!.is_a_string(colour_by) ||
+            !(colour_by %in% names(colData(object)))) ){
+            stop("'colour_by' must be a name of column of colData(object)",
+                 call. = FALSE)
+        }
+        ########################### Input Check end ############################
+        # Gets data that will be plotted. Gets a list
+        density_data_list <- .incorporate_density_data(object = object,
+                                                       abund_values = abund_values,
+                                                       n = n,
+                                                       colour_by = colour_by)
+        # Extracts the density data and aesthetic from the list
+        density_data <- density_data_list$density_data
+        aesthetic <- density_data_list$aesthetic
+        
+        # Gets the plot from plotter
+        plot_out <- .density_plotter(density_data = density_data, 
+                                     aesthetic = aesthetic,
+                                     abund_values = abund_values, 
+                                     colour_by = colour_by)
+        return(plot_out)
+    }
+)
+
+################################ HELP FUNCTIONS ################################
+
+.incorporate_density_data <- function(object, abund_values, n, colour_by){
+    # Gets the assay
+    abundance_table <- assay(object, abund_values, withDimnames=FALSE)
```

Comment:
Using the getter assay `withDimnames = FALSE` is a risk. The subsetting by name in line 105 isn't guaranteed to work in that case, since the assay dimnames maybe set to other values. Have a look at the man page for `assay`

Reply:
Removed